components/director/internal/domain/formation/service.go

```diff
@@ -1529,6 +1529,9 @@ func (s *service) SetFormationToErrorState(ctx context.Context, formation *model
 	formation.Error = marshaledErr
 
 	if err := s.formationRepository.Update(ctx, formation); err != nil {
+		if state == model.DeleteErrorFormationState && apperrors.IsNotFoundError(err) {
```

Comment:
In the cases where you are checking for `DeleteErrorFormationState`, maybe you should also check the cases where the formation is in `DeletingErrorFormationState` also have to be covered. In theory someone could be resynchronizing the formation and resending the notification (it is still in DELETING), but the participant actually responds and deletes it during this operation. Same for each check that is only for `DELETE_ERROR` only.

Reply:
I didn't understand that, here the error state could be either `CreateErrorFormationState` or `DeleteErrorFormationState` based on the operation - `CreateFormation` or `DeleteFormation`? The `DeletingFormationState` is not propagated at all here? Same is valid for the other check in the file